descriptions/rtlpointertooffset.md

```diff
@@ -0,0 +1 @@
+This function is [documented in Windows Driver Kit](https://learn.microsoft.com/en-us/previous-versions/ff553171%28v=vs.85%29).
```

Comment:
Do you have a script that searches for these links? Perhaps it's a good idea to add it to this repo, and run it from time to time to alert about missing links, both due to newly added pages in ntdoc, and due to changes in Microsoft docs.

Reply:
There isn't really a script at this point; the process is still semi-manual. What I did was the following:
1. Clone the main repositories with MS docs.
2. Extract all filenames of markdown files from there.
3. Split them by special characters (mostly dashes) into tokens.
4. Compare these tokens to the list of description IDs.

I suppose we can automate it. For complex IDs it's pretty reliable since if something like `ntopentransactionmanager` appears in a name of a markdown file in MS docs, it's most certainly a match. More simple names, on the other hand, can be a problem.

I processed the following repositories this way:
 - https://github.com/MicrosoftDocs/win32
 - https://github.com/MicrosoftDocs/sdk-api
 - https://github.com/MicrosoftDocs/windows-driver-docs-ddi
 - https://github.com/MicrosoftDocs/windows-driver-docs
 - https://github.com/MicrosoftDocs/windows-itpro-docs
 - https://github.com/MicrosoftDocs/windowsserverdocs
 - https://github.com/MicrosoftDocs/winrt-api
 - https://github.com/MicrosoftDocs/winrt-related
 - https://github.com/MicrosoftDocs/windows-dev-docs
 - https://github.com/MicrosoftDocs/msix-docs

I also tried running full-text search, but it takes too long and yields a lot of false-positive, especially on the pages that include stack traces,

Reply:
I assume links like this were handled with a more manual approach?
**ALIAS_ADD_MEMBER**
https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-samr/2da21c6c-5b15-46c8-bd4e-6a8443216e1a

Reply:
Oh yeah, I forgot to mention specifications, they are a special case. Here I just parsed the table of content from the left side of the page.

Reply:
I thought I could quickly slap something together, but there are more special cases than I thought. There are also the [previous versions](https://github.com/m417z/ntdoc/pull/7#issuecomment-1733629020) docs.  I won't be working on it for now. If you'd like to create such a script, I'll integrate it into the repo as a scheduled check. It can be in any language and doesn't have to be fast. It's enough to have something that receives an identifier and returns/prints the relevant doc links.